### ISSUE TITLE
Appium IOS tests running when server started manually but not when server is started via AppiumDriverLocalService

### ISSUE THREAD

**Issue by fida10** [appium/appium#15119]
## The problem

Attempting to run Appium automation scripts on iOS (simulator) on a Mac Mini (M1 chip, if that's relevant). When I run the tests, they work just fine when the Appium server is started manually (typing "appium" into the terminal and starting it that way). However, when I attempt to start the appium server programmatically, the application under test fails to launch, with the following error:

2021-02-17 03:41:27:256 [W3C]       WebDriverAgentRunner-Runner.app (19077) encountered an error (Failed to load the test bundle. If you believe this error represents a bug, please attach the result bundle at /Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-gkbkvswlszzhhbevpokpwtrjdxxq/Logs/Test/Test-WebDriverAgentRunner-2021.02.16_22-41-22--0500.xcresult. (Underlying Error: **The bundle “WebDriverAgentRunner” couldn’t be loaded because it doesn’t contain a version for the current architecture. The bundle doesn’t contain a version for the current architecture. Try installing a universal version of the bundle.** dlopen_preflight(/Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-gkbkvswlszzhhbevpokpwtrjdxxq/Build/Products/Debug-iphonesimulator/WebDriverAgentRunner-Runner.app/PlugIns/WebDriverAgentRunner.xctest/WebDriverAgentRunner): no suitable image found.  Did find:
2021-02-17 03:41:27:256 [W3C]   /Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-gkbkvswlszzhhbevpokpwtrjdxxq/Build/Products/Debug-iphonesimulator/WebDriverAgentRunner-Runner.app/PlugIns/WebDriverAgentRunner.xctest/WebDriverAgentRunner: mach-o, but wrong architecture))

## Environment

* Appium version (or git revision) that exhibits the issue: 7.4.1
* Desktop OS/version used to run Appium: Mac OS 11.1 (Big Sur)
* Node.js version (unless using Appium.app|exe): Node.js v15.8.0
* Npm or Yarn package manager: npm 7.5.0
* Mobile platform/version under test: iOS Simulator (iPhone SE 2nd Generation 14.4)
* Real device or emulator/simulator: Simulator

## Details

I originally thought that this was an issue with xcode or WDA, but if that were the case, the tests would fail no matter how Appium was started, and as mentioned previously, the tests pass perfectly fine when Appium is started manually (via terminal), so it might be an issue with the PATH or environment variables upon execution from Java, not sure though.

## Link to Appium logs

Here's the full appium log: https://gist.githubusercontent.com/fida10/44344b223874310cf296d38a95d4268f/raw/316855b619129680eeaa6519a446a436d0699cd6/failedLog.txt

## Code To Reproduce Issue [ Good To Have ]

Here is the code I am using to start the server programmatically:

        HashMap<String, String> environment = new HashMap();
        environment.put("PATH", "/usr/local/bin:" + System.getenv("PATH"));

        AppiumDriverLocalService server = AppiumDriverLocalService
                .buildService(new AppiumServiceBuilder()
                        .withEnvironment(environment)
                        .usingDriverExecutable(new File("//opt/homebrew/Cellar/node/15.8.0/bin/node"))
                        .withAppiumJS(new File("//Users/sagolGoru20/.npm-packages/lib/node_modules/appium/build/lib/main.js"))
                        .usingAnyFreePort()
                        .withArgument(GeneralServerFlag.SESSION_OVERRIDE)
                        .withLogFile(new File("//Users/sagolGoru20/Programming/JavaProjects/MavenProjects/MobileAutomationProjects/firstAppiumProject/appiumLog.txt"))
                );
        server.start();
As the appium log shows, the server seems to be started fine but then throws the above error, so I think the issue may be with how I'm building the AppiumDriverLocalService object.

I followed this tutorial in building AppiumDriverLocalService: https://appiumpro.com/editions/71-starting-an-appium-server-programmatically-using-appiumservicebuilder

Any help would be greatly appreciated.

**Comment by mykola-mokhnach** [comment]
Try to cleanup the WebDriverAgent's derived data/clean the project itself. Possible duplicate of https://github.com/appium/appium/issues/15084

**Comment by fida10** [comment]
How do you mean? Here is the project code: 

https://gist.github.com/fida10/bec187a516fc32f907f97725263a7206

It's a simple project, I'm just clicking on an "Allow" button (line 26).

When I comment out the AppiumDriverLocalService server object (lines 46 to 58) and instead launch by uncommenting line 60, the test runs properly.

**Comment by mykola-mokhnach** [comment]
run `/Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-*` and try again

Have you tried the solutions from the linked ticket?

**Comment by fida10** [comment]
I gave those a try, tried reinstalling Appium server and the like, but no luck. I ran the above command you sent just now and it gives me the following: /Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-gkbkvswlszzhhbevpokpwtrjdxxq: is a directory

Here is what is in the directory:

![Screen Shot 2021-02-17 at 1 09 36 PM](https://user-images.githubusercontent.com/3886263/108248206-8db15280-7121-11eb-8919-65d9fa6a81f4.png)

**Comment by mykola-mokhnach** [comment]
simply do `rm -rf /Users/sagolGoru20/Library/Developer/Xcode/DerivedData/WebDriverAgent-*`

**Comment by fida10** [comment]
Ok, I the file was removed; unfortunately I'm still getting the same error message as before, **The bundle “WebDriverAgentRunner” couldn’t be loaded because it doesn’t contain a version for the current architecture. The bundle doesn’t contain a version for the current architecture. Try installing a universal version of the bundle.**

**Comment by mykola-mokhnach** [comment]
How about https://github.com/appium/appium/issues/14102#issuecomment-776616267 ?

**Comment by fida10** [comment]
I gave that one a try as well just now, changed the build settings to validate the workspace, but that did not solve the issue either: 

![Screen Shot 2021-02-17 at 2 05 40 PM](https://user-images.githubusercontent.com/3886263/108254665-5050c300-7129-11eb-85d8-279c8c834d84.png)

I'm going through the new appiumLog now to see if there's anything different there but so far it looks identical more or less to the old log, with the same error. New log: https://gist.githubusercontent.com/fida10/2cfe71774f95106344bdb352ac5081ef/raw/2150eaaea140ebd569ac1a5511727040054be458/appiumLog.txt

Thanks so far for your help...what I'm not getting is why is it working properly when I start the appium server manually, but not working when it is being started programmatically with AppiumDriverLocalService. It used to run just fine on my old intel mac, am seeing this issue only after switching to M1.

**Comment by fida10** [comment]
For comparison, this is the log from when the test succeeded with appium started manually, if you have any insight: https://gist.github.com/fida10/f4bf6f45e8d0f9012ce37249bee1b1b9

**Comment by mykola-mokhnach** [comment]
I would also try https://github.com/microsoft/appcenter/issues/2062

Perhaps the env set by AppiumDriverLocalService has some special stuff. This could be checked in runtime though. See https://github.com/appium/java-client/blob/master/docs/environment.md for more details

**Comment by fida10** [comment]
I think that may be the case; issue may be with environment variables. 

I started a few appium server instances programmatically, and then started appium manually, then used "ps -e|grep node" to see them both. Here were the results: 

```
os ~ $ ps -e|grep node
 3166 ??         0:04.58 /opt/homebrew/Cellar/node/15.8.0/bin/node /Users/sagolGoru20/.npm-packages/lib/node_modules/appium/build/lib/main.js --port 28959 --address 0.0.0.0 --session-override
 4890 ??         0:00.51 /opt/homebrew/Cellar/node/15.8.0/bin/node /Users/sagolGoru20/.npm-packages/lib/node_modules/appium/build/lib/main.js --port 28709 --address 0.0.0.0 --session-override
 5323 ttys002    0:00.01 grep node
os ~ $ killall node
os ~ $ killall node
No matching processes belonging to you were found
os ~ $ ps -e|grep node
 5336 ttys002    0:00.00 grep node
os ~ $ ps -e|grep node
 5446 ttys002    0:00.00 grep node
 5444 ttys003    0:00.40 node /Users/sagolGoru20/.npm-packages/bin/appium
```

The bottom appium instance (ID 5444, started manually) is starting from a completely different place than the top two appium instances (ID 3166, ID 4890, started programmatically with AppiumDriverLocalService. Do you know if this is supposed to happen? If not, any suggestions on how I might fix it? I'm assuming that the /Users/sagolGoru20/.npm-packages/bin/appium from 5444 is the correct path that appium should start from on my device.
